src/controllers/things_controller.js

```diff
@@ -296,7 +303,10 @@ ThingsController.ws('/:thingId/', function(websocket, request) {
           const actionParams = request.data[actionName].input;
           Things.getThing(thingId).then(thing => {
             let action = new Action(actionName, actionParams, thing);
-            return Actions.add(action);
+            return Actions.add(action).then(() => {
+              AddonManager.requestAction(
```

Comment:
I believe that this should be `return`ed so that any errors bubble into the `websocket.send` error instead of being an unhandled promise rejection.

Reply:
Good catch.